src/clients/Tester.ts

```diff
@@ -0,0 +1,90 @@
+import { AppClient, inflightURL, InstanceOptions, IOContext } from '@vtex/api'
+
+export interface SpecTestReport {
+  testId: string
+  title: string[]
+  state: string
+  body: string
+  stack?: string
+  error?: string
+}
+
+export interface Screenshot {
+  screenshotId: string
+  name?: string
+  testId: string
+  takenAt: string
+  path: string
+  height: number
+  width: number
+}
+
+export interface SpecReport {
+  state: 'enqueued' | 'running' | 'passed' | 'failed' | 'skipped' | 'error'
+  error?: string
+  report?: {
+    stats: {
+      suites: number
+      tests: number
+      passes: number
+      pending: number
+      skipped: number
+      failures: number
+    }
+    tests: SpecTestReport[]
+    video?: string
+    screenshots: Screenshot[]
+  }
+  logId?: string
+  lastUpdate: number
+}
+
+export interface AppReport {
+  [spec: string]: SpecReport
+}
+
+export interface TestReport {
+  [appId: string]: AppReport
+}
+
+export interface TestOptions {
+  monitoring: boolean
+  integration: boolean
+  authToken?: string
+  appKey?: string
+  appToken?: string
+}
+
+type Spec = string
+
+export interface TestRequest {
+  testId: string
+  options: TestOptions
+  testers: {
+    [tester: string]: {
+      [appId: string]: Spec[]
+    }
+  }
+  requestedAt: number
+}
+
+export class Tester extends AppClient {
+  constructor(context: IOContext, options?: InstanceOptions) {
+    super('vtex.tester-hub@0.x', context, options)
+  }
+
+  public report(testId: string) {
+    return this.http.get<TestReport>(`/_v/report/${testId}`, {
+      inflightKey: inflightURL,
+      metric: 'tester-report',
+      cacheable: 0,
+    })
+  }
+
+  public test(options: TestOptions, appId = '') {
+    return this.http.post<TestRequest>(`/_v/test/${appId}`, options, {
+      inflightKey: inflightURL,
```

Comment:
The `inflightKey` key is used in case many concurrent requests to this endpoint are made. If they represent the same request, instead of creating N requests, only one request is made that responds to all N calls. This is used in highly concurrent environments, so I don't think this is necessary for this call - `test` is only called one time per `vtex test e2e` execution.

What do you think?

Reply:
I agree